chips/nrf52/src/uart.rs

```diff
@@ -401,19 +403,18 @@ impl kernel::hil::uart::UART for Uarte {
         let regs = unsafe { &*self.regs };
 
         // truncate rx_len if necessary
-        let mut length = rx_len;
-        if rx_len > rx_buf.len() {
-            length = rx_buf.len();
-        }
+        let mut truncated_length = core::cmp::min(rx_len, rx_buf.len());
+        truncated_length = core::cmp::min(truncated_length, 255);
         //**** Probably need to check what happens when you want to wait for more data than the
         //     rx_buffer is sized to fit. I don't know why you'd do this but whatever.
 
-        self.rx_remaining_bytes.set(length);
+        self.rx_remaining_bytes.set(truncated_length);
```

Comment:
this should be `truncated_length - min(buffer_len, actual len)` i.e, not truncated with 255 otherwise it will not receive all bytes with a buffer bigger than 255!